includes/request.js

```diff
@@ -77,82 +77,92 @@ const errorHelper = async function(err) {
-      retry: maxRetries,
-      // retries for non-response e.g. ETIMEDOUT
-      noResponseRetries: maxRetries,
-      backoffType: 'exponential',
-      httpMethodsToRetry: ['GET', 'HEAD', 'POST'],
-      statusCodesToRetry: [
-        [429, 429],
-        [500, 599]
-      ],
-      shouldRetry: err => {
-        const cfg = retryPlugin.getConfig(err);
-        // cap at max retries regardless of response/non-response type
-        if (cfg.currentRetryAttempt >= maxRetries) {
-          return false;
-        } else {
-          return retryPlugin.shouldRetryRequest(err);
-        }
-      },
-      instance: service.getHttpClient()
-    };
-    retryPlugin.attach(service.getHttpClient());
+    authenticator = new IamAuthenticator(iamAuthOpts);
+  } else if (url.username) {
+    authenticator = new CouchdbSessionAuthenticator({
+      username: decodeURIComponent(url.username),
+      password: decodeURIComponent(url.password)
+    });
+  } else {
+    authenticator = new NoAuthAuthenticator();
+  }
+  const serviceOpts = {
+    authenticator,
+    timeout: opts.requestTimeout,
+    // Axios performance options
+    maxContentLength: -1
+  };
 
-    service.setServiceUrl(actUrl.toString());
-    if (authenticator instanceof CouchdbSessionAuthenticator) {
-      // Awkward workaround for known Couch issue with compression on _session requests
-      // It is not feasible to disable compression on all requests with the amount of
-      // data this lib needs to move, so override the property in the tokenManager instance.
-      authenticator.tokenManager.requestWrapperInstance.compressRequestData = false;
-    }
-    if (authenticator.tokenManager && authenticator.tokenManager.requestWrapperInstance) {
-      authenticator.tokenManager.requestWrapperInstance.axiosInstance.interceptors.response.use(null, errorHelper);
-    }
-    // Add error interceptors to put URLs in error messages
-    service.getHttpClient().interceptors.response.use(null, errorHelper);
+  const service = new CloudantV1(serviceOpts);
+  service.setServiceUrl(actUrl.toString());
+  if (authenticator instanceof CouchdbSessionAuthenticator) {
+    // Awkward workaround for known Couch issue with compression on _session requests
+    // It is not feasible to disable compression on all requests with the amount of
+    // data this lib needs to move, so override the property in the tokenManager instance.
+    authenticator.tokenManager.requestWrapperInstance.compressRequestData = false;
+  }
+  return { service, dbName, actUrl };
+}
 
-    // Add request interceptor to add user-agent (adding it with custom request headers gets overwritten)
-    service.getHttpClient().interceptors.request.use(function(requestConfig) {
-      requestConfig.headers['User-Agent'] = userAgent;
-      return requestConfig;
-    }, null);
+function newClient(rawUrl, opts) {
+  const url = new URL(rawUrl);
+  const { service, dbName, actUrl } = newSimpleClient(url, opts);
```

Comment:
Otherwise we end up doing `new URL(rawUrl)` twice for no reason.
```suggestion
  const { service, dbName, actUrl } = newSimpleClient(rawUrl, opts);
```